src/client.ts

```diff
@@ -500,12 +500,12 @@ export class SubscriptionClient {
       this.clearMaxConnectTimeout();
       this.closedByUser = false;
       this.eventEmitter.emit(this.reconnecting ? 'reconnecting' : 'connecting');
-
       const payload: ConnectionParams = typeof this.connectionParams === 'function' ? this.connectionParams() : this.connectionParams;
```

Comment:
We should probably better specify the `ConnectionParams` TS definition to match this new behavior.